packages/manager/apps/pci-cold-archive/src/pages/containers/archive/Archive.page.tsx

```diff
@@ -110,12 +110,17 @@ export default function ArchivePage() {
     return '';
   }, [lockedUntilDays, error.min, error.max, error.nan]);
 
+  const isDisabled =
+    isPending ||
+    (hasRetention && !lockedUntilDays) ||
+    (hasRetention && lockedUntilDays > maxDayRetention);
```

Comment:
As a minor improvement, and to prevent from multiple assertions over `hasRetention` predicate: 

```suggestion
const isRetentionInvalid = hasRetention && (!lockedUntilDays || lockedUntilDays > maxDayRetention);
const isDisabled = isPending || isRetentionInvalid;
```

Or something like that, but it's up to you 😌